{
	"id": "oval:org.almalinux.alsa:def:20248842",
	"version": "635",
	"class": "patch",
	"metadata": {
		"Title": "ALSA-2024:8842: python3.12-urllib3 security update (Moderate)",
		"reference": [
			{
				"ref_id": "CVE-2024-37891",
				"source": "CVE",
				"ref_url": "https://access.redhat.com/security/cve/CVE-2024-37891"
			},
			{
				"ref_id": "RHSA-2024:8842",
				"source": "RHSA",
				"ref_url": "https://access.redhat.com/errata/RHSA-2024:8842"
			},
			{
				"ref_id": "ALSA-2024:8842",
				"source": "ALSA",
				"ref_url": "https://errata.almalinux.org/8/ALSA-2024-8842.html"
			}
		],
		"Description": "urllib3 is a powerful, user-friendly HTTP client for Python. urllib3 brings many critical features that are missing from the Python standard libraries:   \n• Thread safety.  \n• Connection pooling.  \n• Client-side SSL/TLS verification.  \n• File uploads with multipart encoding.  \n• Helpers for retrying requests and dealing with HTTP redirects.  \n• Support for gzip, deflate, brotli, and zstd encoding.  \n• Proxy support for HTTP and SOCKS.  \n• 100% test coverage.  \n\nSecurity Fix(es):  \n\n  * urllib3: proxy-authorization request header is not stripped during cross-origin redirects (CVE-2024-37891)\n\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.\n",
		"advisory": {
			"from": "packager@almalinux.org",
			"Severity": "Moderate",
			"Rights": "Copyright 2024 AlmaLinux OS",
			"issued": {
				"date": "2024-11-05"
			},
			"updated": {
				"date": "2024-11-05"
			},
			"affected_cpe_list": {
				"Cpe": [
					"cpe:/a:almalinux:almalinux:8",
					"cpe:/a:almalinux:almalinux:8::appstream",
					"cpe:/a:almalinux:almalinux:8::powertools",
					"cpe:/a:almalinux:almalinux:8::highavailability",
					"cpe:/a:almalinux:almalinux:8::nfv",
					"cpe:/a:almalinux:almalinux:8::realtime",
					"cpe:/a:almalinux:almalinux:8::resilientstorage",
					"cpe:/a:almalinux:almalinux:8::sap",
					"cpe:/a:almalinux:almalinux:8::sap_hana",
					"cpe:/a:almalinux:almalinux:8::supplementary",
					"cpe:/o:almalinux:almalinux:8",
					"cpe:/o:almalinux:almalinux:8::baseos"
				]
			},
			"bugzilla": [
				{
					"href": "https://bugzilla.redhat.com/2292788",
					"id": "2292788"
				}
			],
			"cve": [
				{
					"text": "CVE-2024-37891",
					"public": "20240617",
					"href": "https://access.redhat.com/security/cve/CVE-2024-37891",
					"impact": "Moderate",
					"cwe": "CWE-669",
					"cvss3": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N"
				}
			]
		}
	},
	"criteria": {
		"operator": "OR",
		"criterias": [
			{
				"operator": "AND",
				"criterias": [
					{
						"operator": "OR",
						"criterions": [
							{
								"test_ref": "oval:org.almalinux.alsa:tst:20248842002",
								"comment": "python3.12-urllib3 is signed with AlmaLinux OS 8 key (51d6647ec21ad6ea)"
							},
							{
								"test_ref": "oval:org.almalinux.alsa:tst:20248842003",
								"comment": "python3.12-urllib3 is signed with AlmaLinux OS 8 key (2ae81e8aced7258b)"
							}
						]
					}
				],
				"criterions": [
					{
						"test_ref": "oval:org.almalinux.alba:tst:20191992003",
						"comment": "AlmaLinux 8 is installed"
					},
					{
						"test_ref": "oval:org.almalinux.alsa:tst:20248842001",
						"comment": "python3.12-urllib3 is earlier than 0:1.26.19-1.el8_10"
					}
				]
			}
		],
		"criterions": [
			{
				"test_ref": "oval:org.almalinux.alba:tst:20191992005",
				"comment": "AlmaLinux must be installed"
			}
		]
	}
}
